[CI] Unit job in CI test failing

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Kubernetes Version

This is the failed action [run](https://github.com/shipwright-io/operator/actions/runs/22130376447/job/64657021190?pr=278) triggered by the [PR ](https://github.com/shipwright-io/operator/pull/278)

But the same Action run on the same PR ran successfully 10 days ago [here](https://github.com/shipwright-io/operator/actions/runs/21739590003/job/62980105149)

main reason behind this is `exec: "etcd": executable file not found in $PATH`

### Shipwright Version

_No response_

### Current Behavior

_No response_

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Anything else?

_No response_

Hey @IrvingMg i have made a fix 

I bumped the `controller-gen/envtest` from `v0.15.0` to `v0.18.0` 
which solved this issue,
>../../../go/pkg/mod/golang.org/x/tools@v0.20.0/internal/tokeninternal/tokeninternal.go:64:9: invalid array length -delta * delta (constant -256 of type int64)
make: *** [Makefile:208: controller-gen] Error 1

added a flag `--use-deprecated-gcs=false` tells the discovery of the binaries to not use the legacy API's

Also bumped the setup-envtest from `release-0.18` to `release-0.22` that's the [latest](https://github.com/kubernetes-sigs/controller-runtime/tree/main/tools/setup-envtest#envtest-binaries-manager)

after making these changes `make test` finally worked without any errors.

--- a/Makefile
+++ b/Makefile
@@ -217,7 +217,7 @@ kustomize: ## Download kustomize locally if necessary.
 ENVTEST = $(shell pwd)/bin/setup-envtest
 .PHONY: envtest
 envtest: ## Download envtest-setup locally if necessary.
-	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@release-0.18)
+	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@release-0.22)
 
 
 OPERATOR_SDK = $(shell pwd)/bin/operator-sdk

